code/data_structures/src/list/singly_linked_list/operations/unclassified/union_intersection_in_list.java

```diff
@@ -0,0 +1,131 @@
+class LinkedList
+{
+    Node head;  // head of list
+ 
+    /* Linked list Node*/
+    class Node
+    {
+        int data;
+        Node next;
+        Node(int d)
+        {
+            data = d;
+            next = null;
+        }
+    }
+ 
+    /* Function to get Union of 2 Linked Lists */
+    void getUnion(Node head1, Node head2)
```

Comment:
I actually have a big design issue in this class. My view is that you shouldn't need to pass the head of a linked list to modify it, since this should be implicit. (All the functions that require `head` should use the member variable). The function here should take just _one_ `LinkedList` and the other should be the class itself. Remember, these are member functions, not standalone functions.